Sources/NIOTestUtils/NIOThreadPoolTaskExecutor.swift

```diff
@@ -0,0 +1,92 @@
+#if compiler(>=6)
+
+import NIOPosix
+
+/// Run a `NIOThreadPool` based `TaskExecutor` while executing the given `body`.
+///
+/// Example usage:
+/// ```swift
+///     await withNIOThreadPoolTaskExecutor(numberOfThreads: 1) { taskExecutor in
+///         Task(executorPreference: taskExecutor) { ... }
+///     }
+/// ```
+///
+/// - warning: Do not escape the task executor from the closure for later use.
+///            This function does not return until the task executor is deinitialized.
+///            Any task using the executor will keep the executor alive. In other words, this
+///            function will not return until all tasks running on the executor are finished.
+///
+/// - warning: In case of cancellation, this function will wait for all tasks running on the
+///            executor to finish. Note that cancellation does not propagate to unstructured tasks.
+///
+/// - Parameters:
+///   - numberOfThreads: The number of threads in the pool.
+///   - body: The closure that will accept the task executor.
+///
+/// - Throws: When `body` throws.
+///
+/// - Returns: The value returned by `body`.
+///
+public func withNIOThreadPoolTaskExecutor<T, Failure>(
+    numberOfThreads: Int,
+    body: (any TaskExecutor) async throws(Failure) -> T
```

Comment:
No need for an `any` here, this can be a `some`.

Reply:
`some` isn't possible here, but I changed it to the concrete type.